### ISSUE TITLE
AttributeError: 'list' object has no attribute 'get'

### ISSUE THREAD

**Issue by mmargegaj** [nltk/nltk#2579]
The issue happens in the nltk/tag/sequential.py file when trying to tag a list of words:
Stacktrace:
File "/home/mario/Desktop/Projects/AlbanianNLP/simple_test.py", line 10, in main
    tagged_words = unitager.tag(words)
  File "/usr/local/lib/python3.8/dist-packages/nltk/tag/sequential.py", line 63, in tag
    tags.append(self.tag_one(tokens, i, tags))
  File "/usr/local/lib/python3.8/dist-packages/nltk/tag/sequential.py", line 83, in tag_one
    tag = tagger.choose_tag(tokens, index, history)
  File "/usr/local/lib/python3.8/dist-packages/nltk/tag/sequential.py", line 142, in choose_tag
    return self._context_to_tag.get(context)
AttributeError: 'list' object has no attribute 'get'

**Comment by tomaarsen** [comment]
What you have likely done is passed a list as the optional `model` parameter of `nltk.UnigramTagger`, while it expected a dictionary instead. This is not a bug, this is accidentally misusing the `nltk.UnigramTagger`. By the way, for future reference, you should include a snippet of the code that failed too, so people can see what's going on.

See this following snippet for a bit more information:
```python
import nltk

words = ["the", "of", "and", "in"]
# Will throw AttributeError: 'list' object has no attribute 'get'
unitagger = nltk.UnigramTagger(train=None, model=["Nonempty List"], backoff=None, cutoff=0, verbose=False)
tagged_words = unitagger.tag(words)

# Will *not* throw an error
unitagger = nltk.UnigramTagger(train=None, model={"Nonempty": "Dict"}, backoff=None, cutoff=0, verbose=False)
tagged_words = unitagger.tag(words)
```

If this was *not* the issue, and there is an actual bug, please post the section of code that you ran to get this error.
But I feel like this can be closed now.

**Comment by mmargegaj** [comment]
I use a list of dictionaries as a model. So you suggest that I should use just a dict like the following format?
{ 
  'word1': 'tag1', 
  'word2': 'tag2'
}

**Comment by tomaarsen** [comment]
@margegaj24 
There are two possible things you can pass to a `UnigramTagger`, as can be seen in the documentation here: http://www.nltk.org/api/nltk.tag.html.
* train `(list(list(tuple(str, str))))` – The corpus of training data, a list of tagged sentences
An example of this can be found here: https://www.nltk.org/book/ch05.html#unigram-tagging. Here, `brown_tagged_sents` looks like the following:
```python
[[('The', 'AT'), ('Fulton', 'NP-TL'), ('County', 'NN-TL'), ('Grand', 'JJ-TL'), ('Jury', 'NN-TL'), ('said', 'VBD'), ('Friday', 'NR'), ('an', 'AT'), ('investigation', 'NN'), ('of', 'IN'), ("Atlanta's", 'NP$'), ('recent', 'JJ'), ('primary', 'NN'), ('election', 'NN'), ('produced', 'VBD'), ('``', '``'), ('no', 'AT'), ('evidence', 'NN'), ("''", "''"), ('that', 'CS'), ('any', 'DTI'), ('irregularities', 'NNS'), ('took', 'VBD'), ('place', 'NN'), ('.', '.')], [('The', 'AT'), ('jury', 'NN'), ('further', 'RBR'), ('said', 'VBD'), ('in', 'IN'), ('term-end', 'NN'), ('presentments', 'NNS'), ('that', 'CS'), ('the', 'AT'), ('City', 'NN-TL'), ('Executive', 'JJ-TL'), ('Committee', 'NN-TL'), (',', ','), ('which', 'WDT'), ('had', 'HVD'), ('over-all', 'JJ'), ('charge', 'NN'), ('of', 'IN'), ('the', 'AT'), ('election', 'NN'), (',', ','), ('``', '``'), ('deserves', 'VBZ'), ('the', 'AT'), ('praise', 'NN'), ('and', 'CC'), ('thanks', 'NNS'), ('of', 'IN'), ('the', 'AT'), ('City', 
'NN-TL'), ('of', 'IN-TL'), ('Atlanta', 'NP-TL'), ("''", "''"), ('for', 'IN'), ('the', 'AT'), ('manner', 'NN'), ('in', 'IN'), ('which', 'WDT'), ('the', 'AT'), ('election', 'NN'), ('was', 'BEDZ'), ('conducted', 'VBN'), ('.', '.')], ...]
```
* model `(dict)` – The tagger model
An example of this can be found here: https://www.nltk.org/book/ch05.html#the-lookup-tagger. Here, `likely_tags` looks like the following:
```python
{'the': 'AT', ',': ',', '.': '.', 'of': 'IN', 'and': 'CC', 'to': 'TO', 'a': 'AT', 'in': 'IN', 'for': 'IN', 'The': 'AT', 'that': 'CS', '``': '``', 'is': 'BEZ', 'was': 'BEDZ', "''": "''", 'on': 'IN', 'at': 'IN', 'with': 'IN', 'be': 'BE', 'by': 'IN', 'as': 'CS', 'he': 'PPS', 'said': 'VBD', 'his': 'PP$', 'will': 'MD', 'it': 'PPS', 'from': 'IN', 'are': 'BER', ';': '.', 'an': 'AT', 'has': 
'HVZ', '--': '--', 'had': 'HVD', 'who': 'WPS', 'have': 'HV', 'not': '*', 'Mrs.': 'NP', 'were': 'BED', 'this': 'DT', 'which': 'WDT', 'would': 'MD', 'their': 'PP$', 'been': 'BEN', 'they': 'PPSS', 'He': 'PPS', 'one': 'CD', 'I': 'PPSS', 'but': 'CC', 'its': 'PP$', 'or': 'CC', ')': ')', 'more': 'AP', 'Mr.': 'NP', '(': '(', 'up': 'RP', 'all': 'ABN', 'out': 'RP', 'last': 'AP', 'two': 'CD', 'other': 'AP', ':': ':', 'new': 'JJ', 'first': 'OD', 'than': 'IN', 'year': 'NN', 'A': 'AT', 'about': 'IN', 'there': 'EX', 'when': 'WRB', 'home': 'NN', 'after': 'IN', 'In': 'IN', 'also': 'RB', 'It': 'PPS', 'over': 'IN', 'into': 'IN', 'no': 'AT', 'But': 'CC', 'made': 'VBN', 'only': 'AP', 'her': 'PP$', 'years': 'NNS', 'time': 'NN', 'three': 'CD', 'them': 'PPO', 'some': 'DTI', 'can': 'MD', 'him': 'PPO', 'New': 'JJ-TL', 'any': 'DTI', 'state': 'NN', '?': '.', 'President': 'NN-TL', 'before': 'IN', 'week': 'NN', 'could': 'MD', 'under': 'IN', 'against': 'IN', 'we': 'PPSS', 'what': 'WDT'}
```

So, in conclusion, yes you can use a dict in that format as a model.

**Comment by mmargegaj** [comment]
The problem was with the stored tagger. I had stored the tagger as a file using a list of dict as a model parameter. converting that list to dict solved the problem.
Thank you.
